### PR TITLE
Issue #5953: Setup 'mvn compile' and 'mvn package' builds in Jenkins

### PR DESCRIPTION
**Pull request by daniilyar**
Issue #5953: Setup 'mvn compile' and 'mvn package' builds in Jenkins

### CODE REVIEW

**Comment by romani** [comment]
@daniilyar , travis is not happy, still problem with line length

File: Jenkinsfile [hunk 1]
```diff
@@ -1,57 +1,85 @@
...
-        dir("${WORKSPACE}/"){
-          
-          // Running the ls command as a 'no-op' is helpful to make sure that the proper sources were checked out
-          sh "ls -lah"
-          
-          LOGS_PUSH( GREEN("Noop 'ls -lah' task finished") )
+    stage ("Prepare (triggered by hand)") {
+      when { expression { getCause(currentBuild).shortDescription.contains("Started by user") } }
+        steps {
+        echo YELLOW("Triggered by hand, so building for master branch")
```
Comment (romani) [inline Jenkinsfile]: CI build have tendency to fail periodically by reasons unrelated to code changes.
It is always required to have ability to restart PR build by manually.

it should be possible.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,57 +1,85 @@
...
+    stage('Build') {
+      parallel {
+        stage('Compile') {
+          steps {
+            runOnHyper("compile", "s4", "${DEFAULT_MAVEN_OPTS}", "mvn -B compile")
+          }
+        }
+        stage('Package') {
+          steps {
+            runOnHyper("package", "m1", "${DEFAULT_MAVEN_OPTS}", "mvn -B package")
```
Comment (romani) [inline Jenkinsfile]: Lets not bother about pricing while we are so underuse hyper.
use m1 as default for all.

it should be: 
`runOnHyper("package", "mvn -B package")`


somewhere in future, when we come to full quote usage it should be `runOnHyperM1` `runOnHyperS4`.

File: Jenkinsfile (hunk 1 above)
Comment (daniilyar) [reply Jenkinsfile]: @romani, sure this is already supported :) 
To re-start the PR build, there are 2 options:
1. Repo team member writes a key phrase 'test this please' in PR's 'conversation' chat
2. PR author can do 'git commit --amend'. Even without any changes, '--amend' updates the commit hash, so the build will be re-triggered regardless the fact that there are no actual code changes

There are also some other options to re-trigger the failed build, but I decided to not implement them for now to not overcomplicate.

File: Jenkinsfile (hunk 2 above)
Comment (romani) [reply Jenkinsfile]: Please also try to reuse stage name, instead of copy paste.

**Comment by daniilyar** [comment]
@romani I think I fixed both your comments (also CI ran to success). Could you please review again?

File: Jenkinsfile (hunk 2 above)
Comment (daniilyar) [reply Jenkinsfile]: Done

**Pull request merged**